### ISSUE TITLE
matcher thinks that two types with same name from different files are the same type.

### ISSUE THREAD

**Issue by f0mster** [switchupcb/copygen#11]
### Reproduce:

add
```
type StructType struct {
    A string
}
```
to  `examples/automatch/domain/domain.go` and 
```
type StructType struct {
    B string
}
```
`examples/automatch/model/model.go`
than add field 
`FieldTypeA StructType`
to both files.
After codegen you will get 
```
	tA.FieldTypeA = fA.FieldTypeA
	tA.FieldTypeA = fA.FieldTypeA
```

1) you will get two same lines
2) types are not compatible.

**Comment by switchupcb** [comment]
Please use the [issue template](https://github.com/switchupcb/copygen/issues/new/choose) to create issues so I may reproduce any bugs.

**Comment by switchupcb** [comment]
Fixed in #12
